src/cryptoadvance/specter/rpc.py

```diff
@@ -64,24 +79,21 @@ def get_configs(config=None):
         config = get_rpcconfig()
     confs = []
     default = {}
-    if "rpcuser" in config["bitcoin.conf"]:
-        default["user"] = config["bitcoin.conf"]["rpcuser"]
-    if "rpcpassword" in config["bitcoin.conf"]:
-        default["passwd"] = config["bitcoin.conf"]["rpcpassword"]
-    if "rpcconnect" in config["bitcoin.conf"]:
-        default["host"] = config["bitcoin.conf"]["rpcconnect"]
-    if "rpcport" in config["bitcoin.conf"]:
-        default["port"] = int(config["bitcoin.conf"]["rpcport"])
-    if "user" in default and "passwd" in default:
-        if "port" in default: # only one bitcoin-cli makes sense in this case
-            confs.append(default)
-            return confs
-        else:
-            for network in RPC_PORTS:
-                o = {"port": RPC_PORTS[network]}
-                o.update(default)
-                confs.append(o)
-            return confs
+    for network in config["bitcoin.conf"]:
+        if "rpcuser" in config["bitcoin.conf"][network]:
+            default["user"] = config["bitcoin.conf"][network]["rpcuser"]
+        if "rpcpassword" in config["bitcoin.conf"][network]:
+            default["passwd"] = config["bitcoin.conf"][network]["rpcpassword"]
+        if "rpcconnect" in config["bitcoin.conf"][network]:
+            default["host"] = config["bitcoin.conf"][network]["rpcconnect"]
+        if "rpcport" in config["bitcoin.conf"][network]:
+            default["port"] = int(config["bitcoin.conf"][network]["rpcport"])
+        if "user" in default and "passwd" in default:
+            if "port" not in default: # only one bitcoin-cli makes sense in this case
```

Comment:
This will work only once, then it will be populated by mainnet config and all other nodes will not be detected.
Should be
```py
if "port" not in config["bitcoin.conf"]["default"]
```